### DISCUSSION TITLE
Consider using Typecsript in strict mode

### DISCUSSION THREAD

**Discussion by YannHulot** [Etheonor/supanextail#1] [Feedback]
Typescript is where the industry is moving. I build all my projects in Typescript because the benefits outweigh the drawbacks. It helps with refactoring, you also have better suggestions in IDEs and final it makes refactoring a breeze. 

You can also catch bugs earlier and you know what the data structure of a variable/parameter is when using them in your UI. 

I will not consider using a library/template that is not written in Typescript. 

As an alternative, you could have two projects, one in TS and one in JS. If you see that one of the two does not sell well, then you can kill it and focus on your best seller version whether it is JS or TS. This way the market will tell you which one to choose.

**Comment by Etheonor** [comment]
I have benchmarked a few competitors and I don't think I can avoid Typescript ^^ 
I started this week and once I have the basics, I will start switching the repo to Typescript.

**Comment by Etheonor** [comment]
I finally merged my TS branch. I think 95% of the work is done but as I'm new to this I may have made some mistakes in the process. ^^
